extensions/auth/opa/impl/src/main/java/org/apache/polaris/extension/auth/opa/OpaPolarisAuthorizerFactory.java

```diff
@@ -0,0 +1,170 @@
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.io.IOException;
+import java.net.URI;
+import java.time.Duration;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.polaris.core.auth.PolarisAuthorizer;
+import org.apache.polaris.core.auth.PolarisAuthorizerFactory;
+import org.apache.polaris.core.config.RealmConfig;
+import org.apache.polaris.extension.auth.opa.token.BearerTokenProvider;
+import org.apache.polaris.extension.auth.opa.token.FileBearerTokenProvider;
+import org.apache.polaris.extension.auth.opa.token.StaticBearerTokenProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Factory for creating OPA-based Polaris authorizer implementations. */
+@ApplicationScoped
+@Identifier("opa")
+class OpaPolarisAuthorizerFactory implements PolarisAuthorizerFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger(OpaPolarisAuthorizerFactory.class);
+
+  private final OpaAuthorizationConfig opaConfig;
+  private CloseableHttpClient httpClient;
+  private BearerTokenProvider bearerTokenProvider;
+
+  @Inject
+  public OpaPolarisAuthorizerFactory(OpaAuthorizationConfig opaConfig) {
+    this.opaConfig = opaConfig;
+  }
+
+  @PostConstruct
+  public void initialize() {
+    // Validate configuration once during startup
+    opaConfig.validate();
+
+    // Create HTTP client once during startup
+    httpClient = createHttpClient();
+
+    // Setup authentication once during startup
+    setupAuthentication(opaConfig.auth().get());
+  }
+
+  @Override
+  public PolarisAuthorizer create(RealmConfig realmConfig) {
+    // All components are now pre-initialized, just create the authorizer
+    URI policyUri = opaConfig.policyUri().get();
+
+    return new OpaPolarisAuthorizer(policyUri, httpClient, bearerTokenProvider);
+  }
+
+  @PreDestroy
+  public void cleanup() {
+    // Clean up bearer token provider resources
+    if (bearerTokenProvider != null) {
+      try {
+        bearerTokenProvider.close();
+        logger.info("Bearer token provider closed successfully");
+      } catch (Exception e) {
+        // Log but don't throw - we're shutting down anyway
+        logger.warn("Error closing bearer token provider: {}", e.getMessage(), e);
+      }
+    }
+
+    // Clean up HTTP client resources
+    if (httpClient != null) {
+      try {
+        httpClient.close();
+        logger.info("HTTP client closed successfully");
+      } catch (IOException e) {
+        // Log but don't throw - we're shutting down anyway
+        logger.warn("Error closing HTTP client: {}", e.getMessage(), e);
+      }
+    }
+  }
+
+  private CloseableHttpClient createHttpClient() {
+    try {
+      if (opaConfig.http().isEmpty()) {
+        throw new IllegalStateException("HTTP configuration is required");
+      }
+      return OpaHttpClientFactory.createHttpClient(opaConfig.http().get());
```

Comment:
Nit:
```suggestion
      if (opaConfig.http().isEmpty()) {
        throw new IllegalStateException("HTTP configuration is required");
      }
      return OpaHttpClientFactory.createHttpClient(opaConfig.http().orElseThrow(() -> new IllegalStateException("HTTP configuration is required")));
```
(irrelevant if `http()` is not an `Optional, see above)